src/RFC-012x_One_sided_replay_attacks.md

```diff
@@ -0,0 +1,116 @@
+
+## Language
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", 
+"NOT RECOMMENDED", "MAY" and "OPTIONAL" in this document are to be interpreted as described in 
+[BCP 14](https://tools.ietf.org/html/bcp14) (covering RFC2119 and RFC8174) when, and only when, they appear in all capitals, as 
+shown here.
+
+## Disclaimer
+
+This document and its content are intended for information purposes only and may be subject to change or update
+without notice.
+
+This document may include preliminary concepts that may or may not be in the process of being developed by the Tari
+community. The release of this document is intended solely for review and discussion by the community of the
+technological merits of the potential system outlined herein.
+
+## Goals
+
+The aim of this Request for Comment (RFC) is to describe ways we can block replay attacks related to one-sided payments using TariScript.
+
+## Related Requests for Comment
+
+* [RFC-0100: Base Layer](RFC-0120_consensusmd)
+* [RFC-0130: Mining](RFCD-201_TariScript.md)
+
+## Replay attack
+
+Replay attacks are "replaying" old messages to deceive the receiver about the message's authenticity. 
+With TariScript, a vulnerability exists where a replay attack can occur under certain conditions even with the current consensus rules. 
+
+For this attack to work, we need Alice and Charlie to collude to steal some of Bob's funds:
+
+* Alice sends a one-sided transaction to Bob. 
+* Bob spends this UTXO to Charlie. 
+  * Bob has to spend this and only this UTXO alone to Charlie with zero change.
+* Alice sends a new one-sided transaction to Bob using the exact same UTXO as before
+* Alice shares the Blinding factor of the UTXO with Charlie
+* Charlie can now claim this UTXO by replaying his old transaction 
+  * Charlie has the signatures to spend the scripts, sign for the changes, etc. 
+  * Because the previous transaction contains no other inputs, Charlie only has to provide signatures for this one UTXO.
+  * Because there is no change UTXO, Charlie has the keys for all the outputs in the transactions and can thus add another transaction or input
+     /output to make sure the kernel excess signature is unique.
+
+This does not work if Bob includes another UTXO in the transaction to Charlie due to the [script offset]. Although Charlie has the 
+blinding factor, for the one UTXO, he does not have the [script offset]. Charlie can create a new kernel signature unique for blockchain consensus with the blinding factor. Still, because the [script offset] needs to balance as well, and he does not know the private keys for this, he needs
+to use this as is, meaning he needs to use an exact copy of the transaction. If the transaction includes any UTXO that he does not know the 
+blinding factor of, he cannot create a new kernel excess signature. Meaning it won't pass consensus rules. 
+
+## Solutions
+
```

Comment:
Add a comment about how niche this attack is; but is nevertheless, a valid attack.